### PR TITLE
AG-932/AG-958: Add biodomains dataset

### PR DESCRIPTION
**Pull request by JessterB**
Note that this won't pass CI yet, and can't be merged until we resolve the DUC issue around the biodomains source file.

The DUC issue has been resolved and CI is passing now.

### CODE REVIEW

File: config.yaml
Comment (jaclynbeck-sage) [inline config.yaml:13]: Is there a reason that this is an `agora_rename` instead of a `column_rename`? Most other datasets appear to be using `column_rename` to standardize to `ensembl_gene_id` before the transform.

File: test_config.yaml
Comment (jaclynbeck-sage) [inline test_config.yaml:13]: Same question

**Review by jaclynbeck-sage — COMMENTED** [review]
Looks good, just waiting on an answer to my question about column renames before approval.

File: config.yaml
Comment (JessterB) [reply config.yaml:13]: Nope! Just confusion on my part...I've updated the aliasing and transform to use ensembl_gene_id now.

File: test_config.yaml
Comment (JessterB) [reply test_config.yaml:13]: Same answer

**Review by jaclynbeck-sage — APPROVED** [review]
Looks great!

**Pull request merged**
